indico/util/marshmallow.py

```diff
@@ -160,11 +160,15 @@ def _deserialize(self, value, attr, data):
 class PrincipalPermissionList(Field):
     """Marshmallow field for a list of principals and their permissions."""
 
-    def __init__(self, principal_class, **kwargs):
+    def __init__(self, principal_class, explicit_permissions=False, **kwargs):
         self.principal_class = principal_class
+        self.explicit_permissions = explicit_permissions
         super(PrincipalPermissionList, self).__init__(**kwargs)
 
     def _serialize(self, value, attr, obj):
+        if self.explicit_permissions:
+            return [(entry.principal.identifier, sorted(get_principal_permissions(entry, type(obj))))
```

Comment:
no need to call `type(obj)` for  every iteration.. assign `obj_cls = type(obj)` before and just use that ;)